compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -1270,14 +1270,17 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                 if let Err(terr) =
                     self.sub_types(rv_ty, place_ty, location.to_locations(), category)
```

Comment:
sooo..... in `issue-91633`, if `sub_types` fails here, what is `rv_ty` and `place_ty` exactly?

Reply:
rv_ty `&T`
place_ty `&<[T] as std::ops::Index<usize>>::Output`

Reply:
oh haha, I get it now.